DNN Platform/DotNetNuke.Web/Common/DotNetNukeHttpApplication.cs

```diff
@@ -133,6 +133,7 @@ private void Application_Start(object sender, EventArgs eventArgs)
             Globals.DependencyProvider = dependencyProvider;
             dependencyProvider.SetProvider(DependencyInjectionInitialize.BuildServiceProvider());
             ServiceRequestScopeModule.SetServiceProvider(Globals.DependencyProvider);
+            HttpRuntime.WebObjectActivator = new RequestScopeServiceProvider();
```

Comment:
I don't believe this code is correct, unless there is a special web forms technique I am missing. We already have an implementation of the `IServiceProvider` and that is the `LazyServiceProvider`. This code would effectively create 2 references to different `IServiceProvider` implementations. See comments on the `RequesScopeServiceProvider`

```suggestion
            HttpRuntime.WebObjectActivator = dependencyProvider;
```

Reply:
See the comment below. The variable `dependencyProvider` is in this case the global scope.

Reply:
This makes a lot a sense, thanks for explaining

Reply:
To clarify, no need to accept my suggested change. I think creating a special ServiceProvider as you originally recommended makes the most sense